synapse/handlers/user_directory.py

```diff
@@ -306,26 +300,24 @@ def _handle_room_publicity_change(self, room_id, prev_event_id, event_id, typ):
         # being added multiple times. The batching upserts shouldn't make this
         # too bad, though.
         for user_id, profile in iteritems(users_with_profile):
-            yield self._handle_new_user(room_id, user_id, profile)
+            await self._handle_new_user(room_id, user_id, profile)
 
-    @defer.inlineCallbacks
-    def _handle_local_user(self, user_id):
+    async def _handle_local_user(self, user_id):
```

Comment:
this seems to be unused :-S

Reply:
Ah, yeah I meant to mention that. Does it seem OK to remove it here?

Reply:
Talked about this in #synapse-dev:matrix.org and we decided it is fine to remove.